nixos/modules/programs/mininet.nix

```diff
@@ -0,0 +1,40 @@
+# Global configuration for mininet
+# kernel must have NETNS/VETH/SCHED
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg  = config.programs.mininet;
+
+  generatedPath = with pkgs; makeSearchPath "bin"  [
+    iperf ethtool iproute socat
+  ];
+
+  pyEnv = pkgs.python.withPackages(ps: [ ps.mininet-python ]);
+
+  mnexecWrapped = pkgs.runCommand "mnexec-wrapper"
+    { buildInputs = [ pkgs.makeWrapper pkgs.pythonPackages.wrapPython ]; }
+    ''
+      makeWrapper ${pkgs.mininet}/bin/mnexec \
+        $out/bin/mnexec \
+        --prefix PATH : "${generatedPath}"
+
+      makeWrapper ${pyEnv}/bin/mn $out/bin/mn
```

Comment:
```
ln -s ${pyEnv}/bin/mn $out/bin/mn
```

Reply:
but actually since you are passing both to systemPackages, you can add them as well both just there. Then the likekelyhood of collsions is lower, when somebody already has the package in its list.

Reply:
ok. maybe `ln -s ${pyEnv}/bin/mn $out/bin/mn` is still required if the environment also contains other binaries.